### DISCUSSION TITLE
Access a function in nuxt.config for a module

### DISCUSSION THREAD

**Discussion by JulienMaurice** [nuxt/nuxt#17376] [Questions] [answered]
Hi,

I'm making a nuxt 3 module to add headers to pages. I want to be able to give the params trought the `nuxt.config`

```ts
nuxt.options.runtimeConfig.public[key] = { ...options[key] }
```

I've been able to give the options from the setup of `defineNuxtModule` trought the `nuxt.options.runtimeConfig` to my `defineNuxtPlugin` but only when it's a string.

```ts
  const {
    public: { serverHeaders, nuxtHeaders } // <== I can't get the functions
  } = useRuntimeConfig()
```

How could I transmit a function from the options of `defineNuxtmodule` to `defineNuxtPlugin` to beneficiate of nuxtApp context ?

Before, in nuxt 2, we could `addServerMiddleware()` that could have the Req context.
Now in nuxt 3, addServerMiddleware can't be used.

**Comment by Flanker72** [comment, marked as answer]
RuntimeConfig is designed to pass plain serializable data. You may acheive the desired behavior by passing options directly to your module like it's described here: https://v3.nuxtjs.org/guide/going-further/modules#module-anatomy

Module is a kind of external source, it should know nothing about the particular project structure or configuration. So, pass module config directly when applying it - and you'll have an ability to make really usable module, applicable to any project.

**Comment by JulienMaurice** [comment]
The solution I found is to stringify the function to pass it trought de runtimeConfig.
There must be a solution with template too.
